### PR TITLE
Some static analysis improvements

### PR DESCRIPTION
**Pull request by franmomu**

### CODE REVIEW

**Review by VincentLanglet — APPROVED** [review]

**Comment by VincentLanglet** [comment]
How much error for a phpstan or psalm bump ?

File: src/Builder/DatagridBuilder.php [hunk 1]
```diff
@@ -94,7 +94,14 @@ public function addFilter(DatagridInterface $datagrid, ?string $type, FieldDescr
     {
         if (null === $type) {
             $guessType = $this->guesser->guess($fieldDescription);
+            if (null === $guessType) {
+                throw new \InvalidArgumentException(sprintf(
+                    'Cannot guess a type for the field description "%s", You MUST provide a type.',
```
Comment (phansys) [inline src/Builder/DatagridBuilder.php:99]:
```suggestion
                    'Cannot guess a type for the field description "%s", you MUST provide a type.',
```

File: src/Builder/ListBuilder.php [hunk 2]
```diff
@@ -49,6 +49,13 @@ public function buildField(?string $type, FieldDescriptionInterface $fieldDescri
     {
         if (null === $type) {
             $guessType = $this->guesser->guess($fieldDescription);
+            if (null === $guessType) {
+                throw new \InvalidArgumentException(sprintf(
+                    'Cannot guess a type for the field description "%s", You MUST provide a type.',
```
Comment (phansys) [inline src/Builder/ListBuilder.php:54]:
```suggestion
                    'Cannot guess a type for the field description "%s", you MUST provide a type.',
```

File: src/Builder/ShowBuilder.php [hunk 3]
```diff
@@ -52,6 +52,13 @@ public function addField(
     ): void {
         if (null === $type) {
             $guessType = $this->guesser->guess($fieldDescription);
+            if (null === $guessType) {
+                throw new \InvalidArgumentException(sprintf(
+                    'Cannot guess a type for the field description "%s", You MUST provide a type.',
```
Comment (phansys) [inline src/Builder/ShowBuilder.php:57]:
```suggestion
                    'Cannot guess a type for the field description "%s", you MUST provide a type.',
```

File: src/Datagrid/Pager.php [hunk 4]
```diff
@@ -37,15 +37,27 @@ public function countResults(): int
 
     public function getCurrentPageResults(): iterable
     {
-        return $this->getQuery()->execute();
+        $query = $this->getQuery();
+
+        if (null === $query) {
+            throw new \RuntimeException('Uninitialized query');
```
Comment (phansys) [inline src/Datagrid/Pager.php:43]:
```suggestion
            throw new \RuntimeException('Uninitialized query.');
```

File: src/Datagrid/Pager.php [hunk 5]
```diff
@@ -37,15 +37,27 @@ public function countResults(): int
...
+        return $query->execute();
     }
 
     public function init(): void
     {
-        $this->resultsCount = $this->computeResultsCount();
+        $query = $this->getQuery();
+
+        if (null === $query) {
+            throw new \RuntimeException('Uninitialized query');
```
Comment (phansys) [inline src/Datagrid/Pager.php:54]:
```suggestion
            throw new \RuntimeException('Uninitialized query.');
```

**Review by phansys — APPROVED** [review]

**Comment by franmomu** [comment]
> How much error for a phpstan or psalm bump ?

I guess not many, I'll try to bump them in the next PR.

**Pull request merged**